src/build_checklist_indicators.Rmd

```diff
@@ -75,57 +75,209 @@ taxon <- trias::gbif_get_taxa(
   select(one_of(checklist_info))
 ```
 
-Extract taxonomic information from GBIF taxonomic backbone checklist. Only taxa found in GBIF backbone will be taken into account (see GitHub issue [#21](https://github.com/trias-project/pipeline/issues/21))
+Extract taxonomic information from GBIF taxonomic backbone checklist. Only taxa found in GBIF backbone will be taken into account (see GitHub issue [#21](https://github.com/trias-project/pipeline/issues/21)), so only taxa with a `nubKey`
 
 ```{r retrieve_taxonomic_info_backbone}
-print("write here your code, Damiano!")
+taxon %<>% filter(is.finite(nubKey))
+taxon_keys <- taxon %>% select(key) %>% pull(key)
+nub_keys <- taxon %>% distinct(nubKey)
+taxon_backbone_info <-  nub_keys %>%
+  rowwise() %>%
+  do_(interp(~ name_usage(key = .$nubKey,
+                         return = "data")))
+taxon_info <- c("key", "species", "genus", "family", "order",
+                "class", "phylum", "kingdom",
+                "rank", "speciesKey", "taxonomicStatus", 
+                "acceptedKey", "accepted")
+taxon_backbone_info %<>%
+  select(key,one_of(taxon_info))
+combined_taxon_info <- taxon %>% left_join(taxon_backbone_info,
+                                           by = c("nubKey" = "key"))
+head(combined_taxon_info)
 ```
 
+### Distribution
+
 Extract distribution information
 
 ```{r retrieve_distribution_information}
-print("write here your code, Damiano!")
+distribution <- data.frame(key = taxon_keys) %>%
+  rowwise() %>%
+  do_(interp(~ as.data.frame(rgbif::name_usage(key = .$key,
+                                               return = "data",
+                                               data = "distribution"))))
+distribution %<>% select(-source)
```

Comment:
maybe just append the `select` to the previous statement after the `do`?

Reply:
Yes, you are right. Just a reminder of previous testing...
Done here 09850a8